apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+      id +
+      '", "change", function() {\n' +
+      '  console.log("' +
+      id +
+      ' photo selected!");\n' +
+      '  console.log(getImageURL("' +
+      id +
+      '"));\n' +
+      '});\n';
+    return code;
+  }
+
+  insertPhotoselect = () =>
+    this.props.onInsertEvent(this.getPhotoselectEventCode());
+
+  render() {
+    const element = this.props.element;
+    const clickName = 'Change';
+    const clickDesc = 'Triggered when a photo is selected.';
+
+    return (
+      <div id="eventRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow={true}
+        />
+        <EventHeaderRow />
+        <EventRow
+          name={clickName}
+          desc={clickDesc}
+          handleInsert={this.insertPhotoselect}
+        />
+      </div>
+    );
+  }
+}
+
+export default {
+  PropertyTab: PhotoChooserProperties,
+  EventTab: PhotoChooserEvents,
+  themeValues: themeValues.photoSelect,
+
+  create: function() {
+    const element = document.createElement('label');
+    element.setAttribute('class', 'img-upload');
+    element.setAttribute('data-canonical-image-url', 'icon://fa-camera');
+    element.style.margin = '0px';
+    element.style.lineHeight = '1';
+    element.style.overflow = 'hidden';
+    element.style.wordWrap = 'break-word';
+    element.style.borderStyle = 'solid';
+
+    element.style.textRendering = 'optimizeSpeed';
+    elementLibrary.setAllPropertiesToCurrentTheme(
+      element,
+      designMode.activeScreen()
+    );
+    element.style.backgroundSize = 'contain';
+    element.style.backgroundRepeat = 'no-repeat';
+    element.style.backgroundPosition = '50% 50%';
+    element.style.backgroundImage =
+      'url(' +
+      assetPrefix.renderIconToString(
+        element.getAttribute('data-canonical-image-url'),
+        element
+      ) +
+      ')';
+
+    element.style.width = '75px';
+    element.style.height = '50px';
+    const newInput = document.createElement('input');
+    newInput.type = 'file';
+    newInput.accept = 'image/*';
+    newInput.capture = 'camera';
+    newInput.hidden = true;
+    element.appendChild(newInput);
+    return element;
+  },
+  onDeserialize: function(element, updateProperty) {
+    // Disable image upload events unless running
+    $(element).on('click', function(e) {
+      if (!Applab.isRunning()) {
+        element.childNodes[0].disabled = true;
+      } else {
+        element.childNodes[0].disabled = false;
+      }
+    });
+    const url = element.getAttribute('data-canonical-image-url');
+    if (url) {
+      updateProperty(element, 'image', url);
+    }
+    // Set border styles for older projects that didn't set them on create:
+    elementUtils.setDefaultBorderStyles(element);
+    // Set the font family for older projects that didn't set them on create:
+    elementUtils.setDefaultFontFamilyStyle(element);
```

Comment:
I don't think you need to worry about older projects. Since this is a new element, all projects will have these styles already.